JavaScript/JavaScriptSDK.Tests/CheckinTests/SendBuffer.tests.ts

```diff
@@ -0,0 +1,420 @@
+
+                // verify
+                Assert.equal(0, buffer.count(), "buffer should be empty");
+            }
+        });
+
+        this.testCase({
+            name: "ArraySendBuffer: call batchPayloads when a buffer is empty",
+            test: () => {
+                // setup
+                var buffer = this.getArraySendBuffer();
+
+                // act
+                var batch = buffer.batchPayloads();
+
+                // verify
+                Assert.equal(null, batch, "expecting null");
+            }
+        });
+
+        this.testCase({
+            name: "ArraySendBuffer: call batchPayloads when a buffer has one element",
+            test: () => {
+                // setup
+                var buffer = this.getArraySendBuffer();
+
+                // act
+                var payload = "{ test: test }";
+
+                buffer.enqueue(payload);
+                var batch = buffer.batchPayloads();
+
+                // verify
+                Assert.equal("[" + payload + "]", batch, "invalid batch");
+            }
+        });
+
+        this.testCase({
+            name: "ArraySendBuffer: call batchPayloads when a buffer has two elements",
+            test: () => {
+                // setup
+                var buffer = this.getArraySendBuffer();
+
+                // act
+                var payload1 = "{ test: test }";
+                var payload2 = "{ }";
+
+                buffer.enqueue(payload1);
+                buffer.enqueue(payload2);
+                var batch = buffer.batchPayloads();
+
+                // verify
+                Assert.equal("[" + payload1 + "," + payload2 + "]", batch, "invalid batch");
+            }
+        });
+
+        this.testCase({
+            name: "ArraySendBuffer: call batchPayloads when a buffer has two elements - emitLineDelimitedJson",
+            test: () => {
+                // setup
+                var buffer = this.getArraySendBuffer(true);
+
+                // act
+                var payload1 = "{ test: test }";
+                var payload2 = "{ test: test }";
+
+                buffer.enqueue(payload1);
+                buffer.enqueue(payload2);
+                var batch = buffer.batchPayloads();
+
+                // verify
+                Assert.equal(payload1 + "\n" + payload2, batch, "invalid batch");
+            }
+        });
+
+        // SessionStorageSendBuffer tests
+
+        this.testCase({
+            name: "SessionStorageSendBuffer: initialize",
+            test: () => {
+                // setup
+                var buffer = this.getSessionStorageSendBuffer();
+
+                // verify
+                Assert.equal(0, buffer.count(), "new buffer should be empty");
+            }
+        });
+
+        this.testCase({
+            name: "SessionStorageSendBuffer: can enqueue and clear the buffer",
+            test: () => {
+                // setup
+                var buffer = this.getSessionStorageSendBuffer();
```

Comment:
Yeah.. I thought about it and for some reason I've decided not to do that. On second thought, you're right. It will be better to reuse the test logic. 

Done.